src/flow/src/adapter/worker.rs

```diff
@@ -0,0 +1,508 @@
+    state: DataflowState,
+    err_collector: ErrCollector,
+}
+
+impl std::fmt::Debug for ActiveDataflowState<'_> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("ActiveDataflowState")
+            .field("df", &"<Hydroflow>")
+            .field("state", &self.state)
+            .field("err_collector", &self.err_collector)
+            .finish()
+    }
+}
+
+impl Default for ActiveDataflowState<'_> {
+    fn default() -> Self {
+        ActiveDataflowState {
+            df: Hydroflow::new(),
+            state: DataflowState::default(),
+            err_collector: ErrCollector::default(),
+        }
+    }
+}
+
+impl<'subgraph> ActiveDataflowState<'subgraph> {
+    /// Create a new render context, assigned with given global id
+    pub fn new_ctx<'ctx>(&'ctx mut self, global_id: GlobalId) -> Context<'ctx, 'subgraph>
+    where
+        'subgraph: 'ctx,
+    {
+        Context {
+            id: global_id,
+            df: &mut self.df,
+            compute_state: &mut self.state,
+            err_collector: self.err_collector.clone(),
+            input_collection: Default::default(),
+            local_scope: Default::default(),
+        }
+    }
+
+    pub fn set_current_ts(&mut self, ts: repr::Timestamp) {
+        self.state.set_current_ts(ts);
+    }
+
+    /// Run all available subgraph
+    ///
+    /// return true if any subgraph actually executed
+    pub fn run_available(&mut self) -> bool {
+        self.state.run_available_with_schedule(&mut self.df)
+    }
+}
+
+#[derive(Debug)]
+pub struct WorkerHandle {
+    itc_client: Mutex<InterThreadCallClient>,
+}
+
+impl WorkerHandle {
+    /// create task, return task id
+    ///
+    #[allow(clippy::too_many_arguments)]
+    pub async fn create_flow(
+        &self,
+        task_id: FlowId,
+        plan: TypedPlan,
+        sink_id: GlobalId,
+        sink_sender: mpsc::UnboundedSender<DiffRow>,
+        source_ids: &[GlobalId],
+        src_recvs: Vec<broadcast::Receiver<DiffRow>>,
+        expire_when: Option<repr::Duration>,
+        create_if_not_exist: bool,
+        err_collector: ErrCollector,
+    ) -> Result<Option<FlowId>, Error> {
+        let req = Request::Create {
+            task_id,
+            plan,
+            sink_id,
+            sink_sender,
+            source_ids: source_ids.to_vec(),
+            src_recvs,
+            expire_when,
+            create_if_not_exist,
+            err_collector,
+        };
+
+        let ret = self.itc_client.lock().await.call_blocking(req).await?;
+        if let Response::Create {
+            result: task_create_result,
+        } = ret
+        {
+            task_create_result
+        } else {
+            InternalSnafu {
+                reason: format!(
+                    "Flow Node/Worker itc failed, expect Response::Create, found {ret:?}"
+                ),
+            }
+            .fail()
+            .with_context(|_| EvalSnafu {})
+        }
```

Comment:
Introducing `enum-as-inner` can keep thing compact:

```rust
#[derive(Debug, enum_as_inner::EnumAsInner)]
enum Response {
```

```suggestion
        ret.into_create()
            .map_err(|ret| {
                InternalSnafu {
                    reason: format!(
                        "Flow Node/Worker itc failed, expect Response::Create, found {ret:?}"
                    ),
                }
                .build()
            })
            .context(EvalSnafu)?
```